### PR TITLE
More warning fixes

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

File: Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs [hunk 1]
```diff
@@ -94,7 +94,7 @@ public override async Task Open(CancellationToken openCancellationToken)
 
             var now = DateTime.UtcNow;
 
-            var _ = StartStreaming(response, taskCompletionSource, LiveStreamCancellationTokenSource.Token);
+            _ = StartStreaming(response, taskCompletionSource, LiveStreamCancellationTokenSource.Token);
```
Comment (EraYaN) [inline Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs]: Would this work without the assignment?

File: Jellyfin.Server/Program.cs [hunk 2]
```diff
@@ -101,7 +118,7 @@ private static async Task StartApp(StartupOptions options)
             SQLitePCL.Batteries_V2.Init();
 
             // Allow all https requests
-            ServicePointManager.ServerCertificateValidationCallback = new RemoteCertificateValidationCallback(delegate { return true; });
+            ServicePointManager.ServerCertificateValidationCallback = new RemoteCertificateValidationCallback(delegate { return true; } );
```
Comment (EraYaN) [inline Jellyfin.Server/Program.cs]: On an unrelated note, this should be configurable. I believe this is for out going connections, and it's not very good practice to have it as default.

File: Jellyfin.Server/SocketSharp/RequestMono.cs [hunk 3]
```diff
@@ -547,15 +557,23 @@ public override string ToString()
                 }
             }
 
+            private const byte LF = (byte)'\n';
+
+            private const byte CR = (byte)'\r';
```
Comment (EraYaN) [inline Jellyfin.Server/SocketSharp/RequestMono.cs]: Unrelated but, what a mess. The file is even called *Mono.cs. Maybe it's not used?

These values do not deserve constants, since they are by definition constant so they might as well be literals.

File: jellyfin.ruleset [hunk 4]
```diff
@@ -3,11 +3,19 @@
   <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
     <!-- disable warning SA1101: Prefix local calls with 'this.' -->
     <Rule Id="SA1101" Action="None" />
+    <!-- disable warning SA1130: Use lambda syntax -->
+    <Rule Id="SA1130" Action="None" />
```
Comment (EraYaN) [inline jellyfin.ruleset]: This rule seems okay to me actually, but I like lambda's a lot.

Comment (Bond-009) [reply jellyfin.ruleset]: It’s actually for the delegate you commented on earlier.

File: Jellyfin.Server/SocketSharp/RequestMono.cs (hunk 3 above)
Comment (Bond-009) [reply Jellyfin.Server/SocketSharp/RequestMono.cs]: It certainly used, but these changes are out of the scope of this PR.

File: Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs]: You’ll get a warning, that what the discard syntax is for.

File: Jellyfin.Server/Program.cs (hunk 2 above)
Comment (EraYaN) [reply Jellyfin.Server/Program.cs]: But probably also out of scope. It might warrant a `//TODO`

Comment (JustAMan) [reply Jellyfin.Server/Program.cs]: Yeah, this is a bad practice, should be off by default.
Definitely warrants a TODO at least.

File: Jellyfin.Server/Jellyfin.Server.csproj [hunk 5]
```diff
@@ -5,11 +5,14 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>netcoreapp2.1</TargetFramework>
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
   </PropertyGroup>
 
   <PropertyGroup>
     <!-- We need C# 7.1 for async main-->
     <LangVersion>latest</LangVersion>
+    <!-- Disable ducumentation warnings (for now) -->
```
Comment (JustAMan) [inline Jellyfin.Server/Jellyfin.Server.csproj]:
```suggestion
    <!-- Disable documentation warnings (for now) -->
```

File: Jellyfin.Server/SocketSharp/RequestMono.cs [hunk 6]
```diff
@@ -429,13 +439,13 @@ public override long Seek(long d, SeekOrigin origin)
...
                         default:
-                            throw new ArgumentException(nameof(origin));
+                            throw new ArgumentException("Unknown SeekOrigin value", nameof(origin));
                     }
 
                     long virt = real - offset;
                     if (virt < 0 || virt > Length)
                     {
-                        throw new ArgumentException();
+                        throw new ArgumentException("Invalid position", nameof(origin));
```
Comment (JustAMan) [inline Jellyfin.Server/SocketSharp/RequestMono.cs]: I think bad position should be tied to `d`, not `origin` here:
```suggestion
                        throw new ArgumentException("Invalid position", nameof(d));
```

File: Jellyfin.Server/SocketSharp/RequestMono.cs [hunk 7]
```diff
@@ -533,12 +538,17 @@ private class HttpMultipart
...
             public class Element
             {
-                public string ContentType;
-                public string Name;
-                public string Filename;
-                public Encoding Encoding;
-                public long Start;
-                public long Length;
+                public string ContentType { get; set; }
+
```
Comment (JustAMan) [inline Jellyfin.Server/SocketSharp/RequestMono.cs]: do we need empty lines between those? you didn't use them in other files here

File: Jellyfin.Server/SocketSharp/RequestMono.cs [hunk 8]
```diff
@@ -570,18 +588,49 @@ public override string ToString()
...
+                        elem.Filename = StripPath(GetContentDispositionAttributeWithEncoding(header, "filename"));
+                    }
+                    else if (StrUtils.StartsWith(header, "Content-Type:", true))
+                    {
+                        elem.ContentType = header.Substring("Content-Type:".Length).Trim();
+                        elem.Encoding = GetEncoding(elem.ContentType);
+                    }
+                }
+
+                long start = 0;
```
Comment (JustAMan) [inline Jellyfin.Server/SocketSharp/RequestMono.cs]:
```suggestion
                long start = data.Position;
```

File: Jellyfin.Server/SocketSharp/SharpWebSocket.cs [hunk 9]
```diff
@@ -117,6 +111,7 @@ public Task SendAsync(string text, bool endOfMessage, CancellationToken cancella
         public void Dispose()
         {
             Dispose(true);
+            GC.SuppressFinalize(this);
```
Comment (JustAMan) [inline Jellyfin.Server/SocketSharp/SharpWebSocket.cs]: can you explain why?

File: Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs [hunk 10]
```diff
@@ -107,6 +110,7 @@ internal static string CheckBadChars(string name)
                 switch (crlf)
                 {
                     case 0:
+                    {
```
Comment (JustAMan) [inline Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs]: why add braces here? they seem to be excessive for me

Comment (Bond-009) [reply Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs]: Pretty sure the the analyzer wanted it.

File: Jellyfin.Server/SocketSharp/SharpWebSocket.cs (hunk 9 above)
Comment (Bond-009) [reply Jellyfin.Server/SocketSharp/SharpWebSocket.cs]: `Objects that implement the IDisposable interface can call this method from the object's IDisposable.Dispose implementation to prevent the garbage collector from calling Object.Finalize on an object that does not require it. Typically, this is done to prevent the finalizer from releasing unmanaged resources that have already been freed by the IDisposable.Dispose implementation.`
https://docs.microsoft.com/en-us/dotnet/api/system.gc.suppressfinalize?redirectedfrom=MSDN&view=netframework-4.7.2#System_GC_SuppressFinalize_System_Object_

TLDR: Don't queue for finalization when derived classes implement a finalizer.
